pennylane/devices/qubit/preprocess.py

```diff
@@ -133,63 +159,31 @@ def batch_fn(res):
     return tapes, batch_fn
 
 
-def check_validity(tape):
-    """Checks whether the operations and observables in queue are all supported by the device.
-
-    Args:
-        tape (.QuantumTape): tape from which to validate operations and observables
-
-    Raises:
-        DeviceError: if there are operations in the queue or observables that the device does
-            not support
-    """
-    for o in tape.operations:
-        operation_name = o.name
-
-        if not _stopping_condition(o):
-            raise DeviceError(f"Gate {operation_name} not supported on Python Device")
-
-    for o in tape.observables:
-        if isinstance(o, Tensor):
-            for i in o.obs:
-                if not _supports_observable(i.name):
-                    raise DeviceError(f"Observable {i.name} not supported on Python Device")
-        else:
-            observable_name = o.name
-
-            if not _supports_observable(observable_name):
-                raise DeviceError(f"Observable {observable_name} not supported on Python Device")
-
-
-def preprocess(tapes, execution_config=None, max_expansion=10):
+def preprocess(
+    circuits: Tuple[qml.tape.QuantumScript],
```

Comment:
since this is kind of a change to the API, I'm wondering if @trbromley has thoughts on the name change? I'm ok with it, but I hesitate a bit because each of these "circuits" will themselves have a `circuit` property. I'm imagining something scary like:
```python
for circuit in circuits:
    circuit = circuit.circuit
```
😨 😨

Reply:
So `preprocess` is an internal implementation detail for an experimental class.

If anything, this will be used by `DefaultQubit2.preprocess(circuits, execution_config)`, so this is aligning the name with the method it will be used by.

Reply:
I didn't realize [it's also called "circuits"](https://github.com/PennyLaneAI/pennylane/blob/b36054fbe7bf62f1186f14139d08c8d189a5f8cc/pennylane/devices/experimental/device_api.py#L140) in the device API. Aligning sounds good, and those changes were already approved

Reply:
> I didn't realize [it's also called "circuits"](https://github.com/PennyLaneAI/pennylane/blob/b36054fbe7bf62f1186f14139d08c8d189a5f8cc/pennylane/devices/experimental/device_api.py#L140) in the device API. Aligning sounds good, and those changes were already approved

this is _very_ old and predates a lot of major PL changes over the years! So something we can adjust if we need to.

Reply:
Oh ignore my comment, I thought you were referring to the original `Device` class.